solidity/contracts/test/TestTBTCDepositorProxy.sol

```diff
@@ -0,0 +1,171 @@
+    event InitializeDepositReturned(uint256 depositKey);
+
+    event FinalizeDepositReturned(uint256 tbtcAmount, bytes32 extraData);
+
+    function initialize(address _bridge, address _tbtcVault) external {
+        __TBTCDepositorProxy_initialize(_bridge, _tbtcVault);
+    }
+
+    function initializeDepositPublic(
+        IBridgeTypes.BitcoinTxInfo calldata fundingTx,
+        IBridgeTypes.DepositRevealInfo calldata reveal,
+        bytes32 extraData
+    ) external {
+        uint256 depositKey = _initializeDeposit(fundingTx, reveal, extraData);
+        emit InitializeDepositReturned(depositKey);
+    }
+
+    function finalizeDepositPublic(uint256 depositKey) external {
+        (uint256 tbtcAmount, bytes32 extraData) = _finalizeDeposit(depositKey);
+        emit FinalizeDepositReturned(tbtcAmount, extraData);
+    }
+
+    function calculateTbtcAmountPublic(
+        uint64 depositAmountSat,
+        uint64 depositTreasuryFeeSat
+    ) external view returns (uint256) {
+        return _calculateTbtcAmount(depositAmountSat, depositTreasuryFeeSat);
+    }
+}
+
+contract MockBridge is IBridge {
+    using BTCUtils for bytes;
+
+    mapping(uint256 => IBridgeTypes.DepositRequest) internal _deposits;
+
+    uint64 internal _depositTxMaxFee = 1 * 1e7; // 0.1 BTC
+
+    event DepositRevealed(uint256 depositKey);
+
+    function revealDepositWithExtraData(
+        IBridgeTypes.BitcoinTxInfo calldata fundingTx,
+        IBridgeTypes.DepositRevealInfo calldata reveal,
+        bytes32 extraData
+    ) external {
+        bytes32 fundingTxHash = abi
+            .encodePacked(
+                fundingTx.version,
+                fundingTx.inputVector,
+                fundingTx.outputVector,
+                fundingTx.locktime
+            )
+            .hash256View();
+
+        uint256 depositKey = uint256(
+            keccak256(
+                abi.encodePacked(fundingTxHash, reveal.fundingOutputIndex)
+            )
+        );
+
+        require(
+            _deposits[depositKey].revealedAt == 0,
+            "Deposit already revealed"
+        );
+
+        IBridgeTypes.DepositRequest memory request;
+
+        request.depositor = msg.sender;
+        request.amount = uint64(10 * 1e8); // 10 BTC
+        /* solhint-disable-next-line not-rely-on-time */
+        request.revealedAt = uint32(block.timestamp);
+        request.vault = reveal.vault;
+        request.treasuryFee = uint64(1 * 1e8); // 1 BTC
+        request.sweptAt = 0;
+        request.extraData = extraData;
+
+        _deposits[depositKey] = request;
+
+        emit DepositRevealed(depositKey);
+    }
+
+    function sweepDeposit(uint256 depositKey) external {
+        require(_deposits[depositKey].revealedAt != 0, "Deposit not revealed");
+        require(_deposits[depositKey].sweptAt == 0, "Deposit already swept");
+        /* solhint-disable-next-line not-rely-on-time */
+        _deposits[depositKey].sweptAt = uint32(block.timestamp);
```

Comment:
In the integrator tests, we will need tokens to be minted to the depositor. I'm wondering whether we could mint them here in `sweepDeposit` and `finalizeOptimisticMintingRequest` or leave it to the integrator.

If leaving it to the integrator it would be worth making `sweepDeposit` and `finalizeOptimisticMintingRequest` public virtual, so the integrator can reuse the test contracts from this file and just extend it with minting.

Reply:
Leaving this for the integrator will give them more flexibility while writing tests. The second argument is that this abstract contract is unaware of the TBTC token and only pure numbers matter here.